source/dub/dub.d

```diff
@@ -87,6 +87,37 @@ PackageSupplier[] defaultPackageSuppliers()
 	];
 }
 
+/** Returns a registry package supplier according to protocol.
+
+	Allowed protocols are dub+http(s):// and maven+http(s)://.
+*/
+PackageSupplier getRegistryPackageSupplier(string url)
+{
+	if (url.toLower.startsWith("dub+"))
```

Comment:
1) avoid the allocation -> `asLowerCase`
2) Do we even need lowering? 
3) You could make this generic and faster (`startsWith` supports a variadic overload for better performance as the string checking is only done once)

```d
import std.algorithm : map;
import std.meta : aliasSeqOf;
enum needles = [
	["dub+", "RegistryPackageSupplier"],
	["maven+", "M2RegistryPackageSupplier"],
];
switch(url.startsWith(aliasSeqOf!(map!(a => a[0])(needles))))
{
	static foreach (uint i, el; needles)
		case i + 1:
			mixin("return new "~ el[1] ~"(URL(url["~el.length.to!string~"0 .. $]));");
	default:
		return new RegistryPackageSupplier(URL(url));
} 
```

edit: you probably can't use `static foreach` yet as GDC doesn't support this yet, but the "normal" CTFE foreach works as well.

Reply:
done